### PR TITLE
Pairing window

### LINKED ISSUE #5712: Support for on-network pairing on ESP32
#### Problem
I would like a way to start on-network commissioning adverts for ESP32. 

 #### Proposed Solution
Please see attached PR - added a menu option.

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
It's convenient to have a way to trigger on-network commissioning for testing purposes

 #### Summary of Changes
Currently "OpenDefaultPairingWindow" assumes BLE when the device is configured for BLE, softAP otherwise. This change adds the option to start on-network pairing for devices that would normally default to BLE.

This is just triggered by a menu option directly, with no smarts at all with regards to which is the correct decision.

 Fixes #5712

### CODE REVIEW

File: src/app/server/Server.h [hunk 1]
```diff
@@ -54,4 +54,4 @@ enum class ResetAdmins
 /**
  * Open the pairing window using default configured parameters.
  */
-CHIP_ERROR OpenDefaultPairingWindow(chip::ResetAdmins resetAdmins);
+CHIP_ERROR OpenDefaultPairingWindow(chip::ResetAdmins resetAdmins, bool onNetwork = false);
```
Comment (andy31415) [inline src/app/server/Server.h:57]: nit: boolean parameters are generally hard to read. Consider some enum value.

what is onNetwork for ?

would it be something like `kMdnsAdvertise` and `kBleAdvertise` ?

Should we error out if asked to ble advertise but NETWORK_LAYER_BLE is not defined?

**Review by mspang — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @msandstedt ?

**Review by msandstedt — APPROVED** [review]

File: src/app/server/Server.h (hunk 1 above)
Comment (msandstedt) [reply src/app/server/Server.h]: @cecille (and @andy31415),

Before we merge, is this part of the comment resolved?

"Should we error out if asked to ble advertise but NETWORK_LAYER_BLE is not defined?"

I see we have the enum now, but I can't tell from the code whether this other piece is resolved.

Comment (cecille) [reply src/app/server/Server.h]: Ah, sorry - missed this part. Right now this never gets called if BLE isn't defined, so it's not going to cause a regression. But, yes, in general I would like to share more code between the BLE and softAP so I will add this in a new PR.

**Pull request merged**